phoenix-core/src/main/java/org/apache/phoenix/iterate/BaseResultIterators.java

```diff
@@ -496,6 +496,17 @@ public BaseResultIterators(QueryPlan plan, Integer perScanLimit, Integer offset,
             return scans;
     }
 
+    private List<HRegionLocation> getRegionBoundaries(ParallelScanGrouper scanGrouper)
+        throws SQLException{
+        if (scanGrouper instanceof DefaultParallelScanGrouper) {
```

Comment:
These casts should not be necessary. Either add an init method to ParallelScanGrouper or call the init from the implementing class in getRegionBoundaries.